R/zchunk_LB151.ag_MIRCA_ctry_C_GLU_irr.R

```diff
@@ -43,73 +46,138 @@ module_aglu_LB151.ag_MIRCA_ctry_C_GLU_irr_DISABLED <- function(command, ...) {
+    ag_HA_ha_ctry_crop %>%
+      left_join( Yieldratio_ctry_crop[ c( "iso", "GLU", "GTAP_crop", "yieldratio") ], by = c( "iso", "GLU", "GTAP_crop") ) %>%  # Map in the yield ratio
+      mutate(irrshareHA=irrHA/(irrHA + rfdHA)) %>%                                                                              # Compute the share of irrigated harvested area
+      mutate(irrshareProd=(irrshareHA*yieldratio)/((irrshareHA*yieldratio)+(1-irrshareHA))) %>%                                 # Compute the share of irrigated production
+      right_join(L100.LDS_ag_prod_t, by=c("iso", "GLU", "GTAP_crop")) %>%                                                       # Map in the total production
+      mutate(irrProd=value*irrshareProd, rfdProd=value*(1-irrshareProd)) ->                                                     # Compute irrigated and rainfed production
+      ag_Prod_t_ctry_crop
+
+    # Split irrigated and rainfed production into separate dataframes
+    ag_Prod_t_ctry_crop %>%
+      select(iso,GLU,GTAP_crop,rfdProd) ->                                      # Remove extra columns
+      L151.ag_rfdProd_t_ctry_crop
+
+    ag_Prod_t_ctry_crop %>%
+      select(iso,GLU,GTAP_crop,irrProd) ->                                      # Remove extra columns
+      L151.ag_irrProd_t_ctry_crop
+
+    L151.ag_irrHA_ha_ctry_crop %>%
+      add_title("Irrigated harvested area by country, GTAP crop, GLU") %>%
+      add_units("Hectares") %>%
+      add_comments("Uses shares of irrigated area from MIRCA, combined with total harvested area, to calculate irrigated harvested area") %>%
+      add_comments("Data is for a single year (circa 2000)") %>%
       add_legacy_name("L151.ag_irrHA_ha_ctry_crop") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_NO_TEST, FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("aglu/FAO_ag_items_PRODSTAT","L100.LDS_ag_HA_ha","L100.MIRCA_irrHA_ha","L100.MIRCA_rfdHA_ha") ->
       L151.ag_irrHA_ha_ctry_crop
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    L151.ag_rfdHA_ha_ctry_crop %>%
+      add_title("Irrigated harvested area by country, GTAP crop, GLU") %>%
+      add_units("Hectares") %>%
+      add_comments("Uses shares of rainfed area from MIRCA, combined with total harvested area, to calculate rainfed harvested area") %>%
+      add_comments("Data is for a single year (circa 2000)") %>%
       add_legacy_name("L151.ag_rfdHA_ha_ctry_crop") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_NO_TEST, FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("aglu/FAO_ag_items_PRODSTAT","L100.LDS_ag_HA_ha","L100.MIRCA_irrHA_ha","L100.MIRCA_rfdHA_ha") ->
```

Comment:
Two questions: do all the outputs use ALL the inputs? Please check and modify if necessary. Also, remember about `same_precursors_as()` which can save typing and make things clearer.